packages/client/src/BrubeckNode.ts

```diff
@@ -170,12 +170,28 @@ export default class BrubeckNode implements Context {
     async openPublishProxyConnectionOnStreamPart(streamPartId: StreamPartID, nodeId: string): Promise<void> {
         try {
             if (!this.cachedNode || !this.startNodeComplete) {
-                const node = await this.startNode()
-                await node.joinStreamPartAsPurePublisher(streamPartId, nodeId)
+                await this.startNode()
             }
-            await this.cachedNode!.joinStreamPartAsPurePublisher(streamPartId, nodeId)
+            await Promise.all([
+                new Promise<void>((resolve, reject) => {
+                    this.cachedNode!.addPurePublishingAcceptedListener((node, stream) => {
+                        if (node === nodeId && stream === streamPartId) {
+                            resolve()
+                        }
+                    })
+                    this.cachedNode!.addPurePublishingRejectedListener((node, stream) => {
+                        if (node === nodeId && stream === streamPartId) {
+                            reject()
+                        }
+                    })
```

Comment:
To me it seems like this check should be included in the method `joinStreamPartAsPurePublisher`, hiding the implementation details from the "user" of network (i.e. client).

Reply:
Sure, I'll move it to the NetworkNode